apollo/integrations/db/dremio_proxy_client.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Dict, Any, List, Union
+
+from pyarrow.flight import FlightDescriptor, FlightCallOptions
+
+from apollo.integrations.db.base_db_proxy_client import BaseDbProxyClient
+
+from pyarrow import flight, Schema, types as pyarrow_types
+
+_ATTR_CONNECT_ARGS = "connect_args"
+
+
+class DremioProxyClient(BaseDbProxyClient):
+    """
+    Proxy client for Dremio. Credentials are expected to be supplied under "connect_args" and
+    will be passed directly to `pyarrow.flight.connect`, so only attributes supported as parameters by
+    `flight.connect` should be passed.
+    """
+
+    def __init__(self, credentials: Optional[Dict], **kwargs: Any):
+        super().__init__(connection_type="dremio")
+        if not credentials or _ATTR_CONNECT_ARGS not in credentials:
```

Comment:
`connect_args` is intended to be used when we're passing everything in connect_args to a connect method like `connect(**connect_args)`, and the idea is that we can pass new parameters without having to update the agent code
in this case it seems you're getting only `host` and `token` from connect_args, so no need to use it, you can check the code for `Looker` where we're getting `client_id` and `client_secret` directly from credentials
Also, in this case, if `flight.connect` accepts more parameters than host, even when we're not using them now, might be good to keep connect_args and use `flight.connect(**connect_args)` and specify `token` in a separate property in credentials.

makes sense?

Reply:
makes total sense! implemented those changes